src/platform_impl/windows/event_loop.rs

```diff
@@ -1871,20 +1871,27 @@ unsafe extern "system" fn public_window_callback<T: 'static>(
         winuser::WM_SETTINGCHANGE => {
             use crate::event::WindowEvent::ThemeChanged;
 
-            let is_dark_mode = try_dark_mode(window);
-            let mut window_state = subclass_input.window_state.lock();
-            let changed = window_state.is_dark_mode != is_dark_mode;
+            let preferred_theme = subclass_input.window_state.lock().preferred_theme.clone();
 
-            if changed {
-                use crate::window::Theme::*;
-                let theme = if is_dark_mode { Dark } else { Light };
+            if preferred_theme == Theme::System {
+                let is_dark_mode = try_dark_mode(window, preferred_theme);
+                let mut window_state = subclass_input.window_state.lock();
+                let changed = window_state.is_dark_mode != is_dark_mode;
 
-                window_state.is_dark_mode = is_dark_mode;
-                mem::drop(window_state);
-                subclass_input.send_event(Event::WindowEvent {
-                    window_id: RootWindowId(WindowId(window)),
-                    event: ThemeChanged(theme),
-                });
+                if changed {
```

Comment:
Could you 'inline' changed thing?

Reply:
Done! Also implemented your other feedback.